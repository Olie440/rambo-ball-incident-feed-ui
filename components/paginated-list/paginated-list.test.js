import React from 'react';
import { shallow } from 'enzyme';

import PaginatedList from './paginated-list.component';
import MockComponent from '~/__mocks__/mock-component';

describe('PaginatedList', () => {
    let element, props;

    beforeEach(() => {
        props = {
            component: MockComponent,
            pageSize: 3,
            data: [ { id: 1 }, { id: 2 }, { id: 3 }, { id: 4 } ],
            classNames: {
                container: 'container',
                children: 'children',
                controls: {
                    container: 'controls_container',
                    leftButton: 'controls_leftButton',
                    pageText: 'controls_pageText',
                    rightButton: 'controls_rightButton'
                }
            }
        };

        element = shallow(<PaginatedList {...props} />);
    });

    it('renders the correct HTML', () => {
        expect(element).toMatchSnapshot();
    });

    it('renders the correct HTML when no class names are provided', () => {
        props.classNames = {};
        element.setProps(props);
        expect(element).toMatchSnapshot();
    });

    describe('when the right button is clicked', () => {
        beforeEach(() => {
            element.find('.controls_rightButton').simulate('click');
        });

        it('renders the next page', () => {
            expect(element).toMatchSnapshot();
        });

        it('increments the current page in state', () => {
            expect(element.state('currentPage')).toEqual(2);
        });
    });

    describe('when the left button is clicked', () => {
        beforeEach(() => {
            element.setState({ currentPage: 2 });
            element.find('.controls_leftButton').simulate('click');
        });

        it('renders the previous page', () => {
            expect(element).toMatchSnapshot();
        });

        it('decrements the current page in state', () => {
            expect(element.state('currentPage')).toEqual(1);
        });
    });
})