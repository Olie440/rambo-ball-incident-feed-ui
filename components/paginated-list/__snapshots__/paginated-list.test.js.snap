// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatedList renders the correct HTML 1`] = `
<div
  className="container"
>
  <div
    className="controls_container"
  >
    <button
      className="controls_leftButton"
      disabled={true}
      onClick={[Function]}
    >
      &lt;
    </button>
    <span
      className="controls_pageText"
    >
      Page 
      1
       of 
      2
    </span>
    <button
      className="controls_rightButton"
      disabled={false}
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <MockComponent
    className="children"
    id={1}
    key="1"
  />
  <MockComponent
    className="children"
    id={2}
    key="2"
  />
  <MockComponent
    className="children"
    id={3}
    key="3"
  />
</div>
`;

exports[`PaginatedList renders the correct HTML when no class names are provided 1`] = `
<div>
  <div>
    <button
      disabled={true}
      onClick={[Function]}
    >
      &lt;
    </button>
    <span>
      Page 
      1
       of 
      2
    </span>
    <button
      disabled={false}
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <MockComponent
    id={1}
    key="1"
  />
  <MockComponent
    id={2}
    key="2"
  />
  <MockComponent
    id={3}
    key="3"
  />
</div>
`;

exports[`PaginatedList when the left button is clicked renders the previous page 1`] = `
<div
  className="container"
>
  <div
    className="controls_container"
  >
    <button
      className="controls_leftButton"
      disabled={true}
      onClick={[Function]}
    >
      &lt;
    </button>
    <span
      className="controls_pageText"
    >
      Page 
      1
       of 
      2
    </span>
    <button
      className="controls_rightButton"
      disabled={false}
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <MockComponent
    className="children"
    id={1}
    key="1"
  />
  <MockComponent
    className="children"
    id={2}
    key="2"
  />
  <MockComponent
    className="children"
    id={3}
    key="3"
  />
</div>
`;

exports[`PaginatedList when the right button is clicked renders the next page 1`] = `
<div
  className="container"
>
  <div
    className="controls_container"
  >
    <button
      className="controls_leftButton"
      disabled={false}
      onClick={[Function]}
    >
      &lt;
    </button>
    <span
      className="controls_pageText"
    >
      Page 
      2
       of 
      2
    </span>
    <button
      className="controls_rightButton"
      disabled={true}
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <MockComponent
    className="children"
    id={4}
    key="4"
  />
</div>
`;
